diff -ur nginx.1.4.4.orig/src/http/modules/ngx_http_fastcgi_module.c nginx-1.4.4/src/http/modules/ngx_http_fastcgi_module.c
--- nginx.1.4.4.orig/src/http/modules/ngx_http_fastcgi_module.c	2013-11-19 19:25:24.000000000 +0800
+++ nginx-1.4.4/src/http/modules/ngx_http_fastcgi_module.c	2014-02-20 16:01:11.997244659 +0800
@@ -185,6 +185,16 @@
     { ngx_string("invalid_header"), NGX_HTTP_UPSTREAM_FT_INVALID_HEADER },
     { ngx_string("http_500"), NGX_HTTP_UPSTREAM_FT_HTTP_500 },
     { ngx_string("http_503"), NGX_HTTP_UPSTREAM_FT_HTTP_503 },
+    { ngx_string("http_570"), NGX_HTTP_UPSTREAM_FT_HTTP_570 },
+    { ngx_string("http_571"), NGX_HTTP_UPSTREAM_FT_HTTP_571 },
+    { ngx_string("http_572"), NGX_HTTP_UPSTREAM_FT_HTTP_572 },
+    { ngx_string("http_573"), NGX_HTTP_UPSTREAM_FT_HTTP_573 },
+    { ngx_string("http_574"), NGX_HTTP_UPSTREAM_FT_HTTP_574 },
+    { ngx_string("http_575"), NGX_HTTP_UPSTREAM_FT_HTTP_575 },
+    { ngx_string("http_576"), NGX_HTTP_UPSTREAM_FT_HTTP_576 },
+    { ngx_string("http_577"), NGX_HTTP_UPSTREAM_FT_HTTP_577 },
+    { ngx_string("http_578"), NGX_HTTP_UPSTREAM_FT_HTTP_578 },
+    { ngx_string("http_579"), NGX_HTTP_UPSTREAM_FT_HTTP_579 },
     { ngx_string("http_404"), NGX_HTTP_UPSTREAM_FT_HTTP_404 },
     { ngx_string("updating"), NGX_HTTP_UPSTREAM_FT_UPDATING },
     { ngx_string("off"), NGX_HTTP_UPSTREAM_FT_OFF },
diff -ur nginx.1.4.4.orig/src/http/modules/ngx_http_proxy_module.c nginx-1.4.4/src/http/modules/ngx_http_proxy_module.c
--- nginx.1.4.4.orig/src/http/modules/ngx_http_proxy_module.c	2013-11-19 19:25:24.000000000 +0800
+++ nginx-1.4.4/src/http/modules/ngx_http_proxy_module.c	2014-02-20 16:02:39.467254909 +0800
@@ -177,6 +177,16 @@
     { ngx_string("http_500"), NGX_HTTP_UPSTREAM_FT_HTTP_500 },
     { ngx_string("http_502"), NGX_HTTP_UPSTREAM_FT_HTTP_502 },
     { ngx_string("http_503"), NGX_HTTP_UPSTREAM_FT_HTTP_503 },
+    { ngx_string("http_570"), NGX_HTTP_UPSTREAM_FT_HTTP_570 },
+    { ngx_string("http_571"), NGX_HTTP_UPSTREAM_FT_HTTP_571 },
+    { ngx_string("http_572"), NGX_HTTP_UPSTREAM_FT_HTTP_572 },
+    { ngx_string("http_573"), NGX_HTTP_UPSTREAM_FT_HTTP_573 },
+    { ngx_string("http_574"), NGX_HTTP_UPSTREAM_FT_HTTP_574 },
+    { ngx_string("http_575"), NGX_HTTP_UPSTREAM_FT_HTTP_575 },
+    { ngx_string("http_576"), NGX_HTTP_UPSTREAM_FT_HTTP_576 },
+    { ngx_string("http_577"), NGX_HTTP_UPSTREAM_FT_HTTP_577 },
+    { ngx_string("http_578"), NGX_HTTP_UPSTREAM_FT_HTTP_578 },
+    { ngx_string("http_579"), NGX_HTTP_UPSTREAM_FT_HTTP_579 },
     { ngx_string("http_504"), NGX_HTTP_UPSTREAM_FT_HTTP_504 },
     { ngx_string("http_404"), NGX_HTTP_UPSTREAM_FT_HTTP_404 },
     { ngx_string("updating"), NGX_HTTP_UPSTREAM_FT_UPDATING },
diff -ur nginx.1.4.4.orig/src/http/modules/ngx_http_scgi_module.c nginx-1.4.4/src/http/modules/ngx_http_scgi_module.c
--- nginx.1.4.4.orig/src/http/modules/ngx_http_scgi_module.c	2013-11-19 19:25:25.000000000 +0800
+++ nginx-1.4.4/src/http/modules/ngx_http_scgi_module.c	2014-02-20 16:03:35.909118871 +0800
@@ -65,6 +65,16 @@
     { ngx_string("invalid_header"), NGX_HTTP_UPSTREAM_FT_INVALID_HEADER },
     { ngx_string("http_500"), NGX_HTTP_UPSTREAM_FT_HTTP_500 },
     { ngx_string("http_503"), NGX_HTTP_UPSTREAM_FT_HTTP_503 },
+    { ngx_string("http_570"), NGX_HTTP_UPSTREAM_FT_HTTP_570 },
+    { ngx_string("http_571"), NGX_HTTP_UPSTREAM_FT_HTTP_571 },
+    { ngx_string("http_572"), NGX_HTTP_UPSTREAM_FT_HTTP_572 },
+    { ngx_string("http_573"), NGX_HTTP_UPSTREAM_FT_HTTP_573 },
+    { ngx_string("http_574"), NGX_HTTP_UPSTREAM_FT_HTTP_574 },
+    { ngx_string("http_575"), NGX_HTTP_UPSTREAM_FT_HTTP_575 },
+    { ngx_string("http_576"), NGX_HTTP_UPSTREAM_FT_HTTP_576 },
+    { ngx_string("http_577"), NGX_HTTP_UPSTREAM_FT_HTTP_577 },
+    { ngx_string("http_578"), NGX_HTTP_UPSTREAM_FT_HTTP_578 },
+    { ngx_string("http_579"), NGX_HTTP_UPSTREAM_FT_HTTP_579 },
     { ngx_string("http_404"), NGX_HTTP_UPSTREAM_FT_HTTP_404 },
     { ngx_string("updating"), NGX_HTTP_UPSTREAM_FT_UPDATING },
     { ngx_string("off"), NGX_HTTP_UPSTREAM_FT_OFF },
diff -ur nginx.1.4.4.orig/src/http/modules/ngx_http_uwsgi_module.c nginx-1.4.4/src/http/modules/ngx_http_uwsgi_module.c
--- nginx.1.4.4.orig/src/http/modules/ngx_http_uwsgi_module.c	2013-11-19 19:25:25.000000000 +0800
+++ nginx-1.4.4/src/http/modules/ngx_http_uwsgi_module.c	2014-02-20 16:04:25.287244257 +0800
@@ -78,6 +78,16 @@
     { ngx_string("invalid_header"), NGX_HTTP_UPSTREAM_FT_INVALID_HEADER },
     { ngx_string("http_500"), NGX_HTTP_UPSTREAM_FT_HTTP_500 },
     { ngx_string("http_503"), NGX_HTTP_UPSTREAM_FT_HTTP_503 },
+    { ngx_string("http_570"), NGX_HTTP_UPSTREAM_FT_HTTP_570 },
+    { ngx_string("http_571"), NGX_HTTP_UPSTREAM_FT_HTTP_571 },
+    { ngx_string("http_572"), NGX_HTTP_UPSTREAM_FT_HTTP_572 },
+    { ngx_string("http_573"), NGX_HTTP_UPSTREAM_FT_HTTP_573 },
+    { ngx_string("http_574"), NGX_HTTP_UPSTREAM_FT_HTTP_574 },
+    { ngx_string("http_575"), NGX_HTTP_UPSTREAM_FT_HTTP_575 },
+    { ngx_string("http_576"), NGX_HTTP_UPSTREAM_FT_HTTP_576 },
+    { ngx_string("http_577"), NGX_HTTP_UPSTREAM_FT_HTTP_577 },
+    { ngx_string("http_578"), NGX_HTTP_UPSTREAM_FT_HTTP_578 },
+    { ngx_string("http_579"), NGX_HTTP_UPSTREAM_FT_HTTP_579 },
     { ngx_string("http_404"), NGX_HTTP_UPSTREAM_FT_HTTP_404 },
     { ngx_string("updating"), NGX_HTTP_UPSTREAM_FT_UPDATING },
     { ngx_string("off"), NGX_HTTP_UPSTREAM_FT_OFF },
diff -ur nginx.1.4.4.orig/src/http/ngx_http_upstream.c nginx-1.4.4/src/http/ngx_http_upstream.c
--- nginx.1.4.4.orig/src/http/ngx_http_upstream.c	2013-11-19 19:25:25.000000000 +0800
+++ nginx-1.4.4/src/http/ngx_http_upstream.c	2014-02-20 15:58:32.597241451 +0800
@@ -368,6 +368,16 @@
     { 500, NGX_HTTP_UPSTREAM_FT_HTTP_500 },
     { 502, NGX_HTTP_UPSTREAM_FT_HTTP_502 },
     { 503, NGX_HTTP_UPSTREAM_FT_HTTP_503 },
+    { 570, NGX_HTTP_UPSTREAM_FT_HTTP_570 },
+    { 571, NGX_HTTP_UPSTREAM_FT_HTTP_571 },
+    { 572, NGX_HTTP_UPSTREAM_FT_HTTP_572 },
+    { 573, NGX_HTTP_UPSTREAM_FT_HTTP_573 },
+    { 574, NGX_HTTP_UPSTREAM_FT_HTTP_574 },
+    { 575, NGX_HTTP_UPSTREAM_FT_HTTP_575 },
+    { 576, NGX_HTTP_UPSTREAM_FT_HTTP_576 },
+    { 577, NGX_HTTP_UPSTREAM_FT_HTTP_577 },
+    { 578, NGX_HTTP_UPSTREAM_FT_HTTP_578 },
+    { 579, NGX_HTTP_UPSTREAM_FT_HTTP_579 },
     { 504, NGX_HTTP_UPSTREAM_FT_HTTP_504 },
     { 404, NGX_HTTP_UPSTREAM_FT_HTTP_404 },
     { 0, 0 }
diff -ur nginx.1.4.4.orig/src/http/ngx_http_upstream.h nginx-1.4.4/src/http/ngx_http_upstream.h
--- nginx.1.4.4.orig/src/http/ngx_http_upstream.h	2013-11-19 19:25:25.000000000 +0800
+++ nginx-1.4.4/src/http/ngx_http_upstream.h	2014-02-20 15:55:42.217233637 +0800
@@ -23,6 +23,16 @@
 #define NGX_HTTP_UPSTREAM_FT_HTTP_500        0x00000010
 #define NGX_HTTP_UPSTREAM_FT_HTTP_502        0x00000020
 #define NGX_HTTP_UPSTREAM_FT_HTTP_503        0x00000040
+#define NGX_HTTP_UPSTREAM_FT_HTTP_570        0x00000060
+#define NGX_HTTP_UPSTREAM_FT_HTTP_571        0x00000061
+#define NGX_HTTP_UPSTREAM_FT_HTTP_572        0x00000062
+#define NGX_HTTP_UPSTREAM_FT_HTTP_573        0x00000063
+#define NGX_HTTP_UPSTREAM_FT_HTTP_574        0x00000064
+#define NGX_HTTP_UPSTREAM_FT_HTTP_575        0x00000065
+#define NGX_HTTP_UPSTREAM_FT_HTTP_576        0x00000066
+#define NGX_HTTP_UPSTREAM_FT_HTTP_577        0x00000067
+#define NGX_HTTP_UPSTREAM_FT_HTTP_578        0x00000068
+#define NGX_HTTP_UPSTREAM_FT_HTTP_579        0x00000069
 #define NGX_HTTP_UPSTREAM_FT_HTTP_504        0x00000080
 #define NGX_HTTP_UPSTREAM_FT_HTTP_404        0x00000100
 #define NGX_HTTP_UPSTREAM_FT_UPDATING        0x00000200
@@ -34,6 +44,16 @@
 #define NGX_HTTP_UPSTREAM_FT_STATUS          (NGX_HTTP_UPSTREAM_FT_HTTP_500  \
                                              |NGX_HTTP_UPSTREAM_FT_HTTP_502  \
                                              |NGX_HTTP_UPSTREAM_FT_HTTP_503  \
+                                             |NGX_HTTP_UPSTREAM_FT_HTTP_570  \
+                                             |NGX_HTTP_UPSTREAM_FT_HTTP_571  \
+                                             |NGX_HTTP_UPSTREAM_FT_HTTP_572  \
+                                             |NGX_HTTP_UPSTREAM_FT_HTTP_573  \
+                                             |NGX_HTTP_UPSTREAM_FT_HTTP_574  \
+                                             |NGX_HTTP_UPSTREAM_FT_HTTP_575  \
+                                             |NGX_HTTP_UPSTREAM_FT_HTTP_576  \
+                                             |NGX_HTTP_UPSTREAM_FT_HTTP_577  \
+                                             |NGX_HTTP_UPSTREAM_FT_HTTP_578  \
+                                             |NGX_HTTP_UPSTREAM_FT_HTTP_579  \
                                              |NGX_HTTP_UPSTREAM_FT_HTTP_504  \
                                              |NGX_HTTP_UPSTREAM_FT_HTTP_404)
 
